be/src/olap/single_replica_compaction.cpp

```diff
@@ -399,8 +401,16 @@ Status SingleReplicaCompaction::_download_files(DataDir* data_dir,
     MonotonicStopWatch watch;
     watch.start();
     for (auto& file_name : file_name_list) {
-        auto remote_file_url = remote_url_prefix + file_name;
-
+        std::string remote_file_url;
+        // The file name of the variant column with the inverted index contains %
+        // We need to handle %, otherwise it will cause an HTTP 404 error.
+        if (file_name.find('%') != std::string::npos) {
```

Comment:
adding an url example here

Reply:
done

Reply:
what if variant_column_name contains a '@' symbol?

Some users has the following ddl
```sql
CREATE TABLE log_test (
    `@@id` CHAR(34) NOT NULL ,
    `@message` STRING NOT NULL ,

    INDEX idx_message_inv(`@message`) USING INVERTED PROPERTIES(
        "parser" = "unicode",
        "support_phrase" = "true"
    )
)
```

Reply:
'@' is also used as a separator in the variant column's inverted index file name and does not cause any additional impact. such as 020000000000003f624c4c322c568271060f9b5b274a4a95_0_10133@properties%2Emessage.idx, `properties` is a variant column name, `10133` is  a index id.